gettsim/taxes/zve.py

```diff
@@ -313,57 +318,145 @@ def calc_gross_e7(tax_unit, params):
 
 
 def vorsorge2010(tax_unit, params, soz_vers_beitr_params):
-    """'Vorsorgeaufwendungen': Deduct part of your social insurance contributions
-        from your taxable income.
-        This regulation has been changed often in recent years. In order not to make
-        anyone worse off, the old regulation was maintained. Nowadays the older
-        regulations don't play a large role (i.e. the new one is more beneficial most of
-         the times) but they'd need to be implemented if earlier years are modelled.
-        Vorsorgeaufwendungen until 2004
-        TODO
-        Vorsorgeaufwendungen since 2010
+    """ Vorsorgeaufwendungen 2010 regime
         § 10 (3) EStG
         The share of deductable pension contributions increases each year by 2 pp.
         ('nachgelagerte Besteuerung'). In 2018, it's 86%. Add other contributions;
         4% from health contributions are not deductable.
         only deduct pension contributions up to the ceiling. multiply by 2
         because it's both employee and employer contributions.
+
+        Contributions to other security branches can also be deducted.
+
+        This ruling differs to vorsorge2005() only in the treatment of other contributions.
```

Comment:
So are they not close enough to each other to have one function with a case distinction?

Reply:
They are pretty close and hence both use `calc_altersvors_aufwand()`. The case distinction is 'one level above'